### DISCUSSION TITLE
Is it supposed to be compatible with EF7?

### DISCUSSION THREAD

**Discussion by Jogai** [leonardoporro/Detached-Mapper#62] [Q&A] [answered]
After upgrade it seems fine. But I was considering using [unidirectional-many-to-many-relationships](https://learn.microsoft.com/en-us/ef/core/what-is-new/ef-core-7.0/whatsnew#unidirectional-many-to-many-relationships) but I wasn't sure if Detached-Mapper would handle that scenario too.

**Comment by leonardoporro** [comment]
The library just read an entity graph and copies data from DTOs to it (so simple and complicated at the same time).
Having or not a navigation property on the other side I guess it won't affect.. Maybe it's even better.
About ef 7, I haven't it tested yet but it's in the plan..

**Comment by leonardoporro** [comment, marked as answer]
Tested it, and works with both EF6 and EF7 if I keep the minimum EF reference at 6
There are no breaking changes in EF7
New releases of the library will be +7

**Comment by leonardoporro** [comment]
Now it should be with 6, 7, and 8, and each version will use the latest counterpart of EF. 
For example, if you add 8.0.4 package to a net6.0 app, it will depend on EF 6.0.26
But if you add it to a net7.0 package, it will depend on 7.0.15
